src/app/components/ListSelector/ListSelector.jsx

```diff
@@ -4,17 +4,44 @@ import axios from 'axios';
 import ListFilter from './ListFilter.jsx';
 import config from '../../../../appConfig';
 import BookActions from '../../../app/actions/BookActions.js';
+import {
```

Comment:
The whole app should know about the history and now just this component. This should go in utils somewhere or at the top level in `Application.jsx`.